src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/CreateGoalsTest.kt

```diff
@@ -133,146 +149,137 @@ class CreateGoalTest : IntegrationTestBase() {
       .isCreated()
 
     // Then
-    val actionPlan = actionPlanRepository.findByPrisonNumber(prisonNumber)
-    assertThat(actionPlan)
+    val actionPlanResponse = getActionPlan(prisonNumber)
+    assertThat(actionPlanResponse)
       .isForPrisonNumber(prisonNumber)
       .hasNumberOfGoals(1)
-      .wasCreatedBy(dpsUsername)
-    val goal = actionPlan!!.goals!![0]
-    assertThat(goal)
-      .hasTitle(createGoalRequest.title)
-      .hasNumberOfSteps(createGoalRequest.steps.size)
-      .wasCreatedAtPrison(createGoalRequest.prisonId)
-      .wasCreatedBy(dpsUsername)
-      .hasCreatedByDisplayName(displayName)
-      .wasUpdatedAtPrison(createGoalRequest.prisonId)
-      .wasUpdatedBy(dpsUsername)
-      .hasUpdatedByDisplayName(displayName)
-    val step = goal.steps!![0]
-    assertThat(step)
-      .hasTitle(stepRequest.title)
-      .hasTargetDateRange(TargetDateRangeEntity.ZERO_TO_THREE_MONTHS)
-      .hasStatus(StepStatus.NOT_STARTED)
-      .wasCreatedBy(dpsUsername)
+      .goal(1) { goal ->
+        goal.hasTitle(createGoalRequest.title)
+          .hasNumberOfSteps(createGoalRequest.steps.size)
+          .wasCreatedAtPrison(createGoalRequest.prisonId)
+          .wasCreatedBy(dpsUsername)
+          .hasCreatedByDisplayName(displayName)
+          .wasUpdatedAtPrison(createGoalRequest.prisonId)
+          .wasUpdatedBy(dpsUsername)
+          .hasUpdatedByDisplayName(displayName)
+          .step(1) { step ->
+            step.hasTitle(stepRequest.title)
+              .hasTargetDateRange(TargetDateRange.ZERO_TO_THREE_MONTHS)
+              .hasStatus(StepStatus.NOT_STARTED)
+          }
+      }
 
+    val goal = actionPlanResponse.goals[0]
     val expectedEventCustomDimensions = mapOf(
       "status" to "ACTIVE",
       "stepCount" to "1",
-      "reference" to goal.reference.toString(),
+      "reference" to goal.goalReference.toString(),
       "notesCharacterCount" to "23",
     )
     await.untilAsserted {
       verify(telemetryClient).trackEvent("goal-create", expectedEventCustomDimensions, null)
     }
 
     // assert timeline event is created successfully
-    val prisonerTimeline = timelineRepository.findByPrisonNumber(prisonNumber)!!
-    assertThat(prisonerTimeline.prisonNumber).isEqualTo(prisonNumber)
-    val events = prisonerTimeline.events!!
-    assertThat(events.size).isEqualTo(1)
-    assertThat(events[0]).hasEventType(TimelineEventType.ACTION_PLAN_CREATED)
-    assertThat(events[0]).hasSourceReference(actionPlan.reference.toString())
-    assertThat(events[0]).hasNoContextualInfo()
-    assertThat(events[0]).hasAReference()
-    assertThat(events[0]).hasJpaManagedFieldsPopulated()
+    val prisonerTimeline = getTimeline(prisonNumber)
+    assertThat(prisonerTimeline)
+      .isForPrisonNumber(prisonNumber)
```

Comment:
Oops, I forgot to use the assertion for prisonNumber and also about the chaining! 🤦‍♂️